### PR TITLE
Fs 1872 search

### PR DESCRIPTION
**Pull request by adamdavies1**
Adding search and filter criteria to all_assessments_for_fund_round_id route. Creating the sql search for search and filter with test data

### CODE REVIEW

File: config/envs/default.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from config.mappings import assessment_mapping
 from fsd_utils import CommonConfig
 from fsd_utils import configclass
+from fsd_utils import NotifyConstants
```
Comment (adamdavies1) [inline config/envs/default.py:8]: This came from a merge from main. Not sure why its showing here

Comment (srh-sloan) [reply config/envs/default.py:8]: Doesn't look like it's used anywhere so you can probably delete it

File: tests/test_routes.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from db.models.assessment_record.assessment_records import AssessmentRecord
 from tests._helpers import get_random_row
+from tests._helpers import get_rows_by_filters
 
 
 def test_gets_all_apps_for_fund_round(request, client):
```
Comment (srh-sloan) [inline tests/test_routes.py:6]: Can we move away from randomly generated test data please? Makes the tests not repeatable and much harder to debug :)

File: db/queries/assessment_records/queries.py [hunk 3]
```diff
@@ -15,7 +15,13 @@
...
+def get_metadata_for_fund_round_id(
+    fund_id: str,
+    round_id: str,
+    search_term: str,
+    asset_type: str,
+    status: str,
+) -> List[Dict]:
     """get_metadata_for_fund_round_id Executes a query on assessment records
     which returns all rows matching the given fund_id and round_id. Excludes
     irrelevant columns such as `db.models.AssessmentRecord.jsonb_blob`.
```
[the remark this answers is not in the document]
Comment (adamdavies1) [reply db/queries/assessment_records/queries.py]: update this

**Review by srh-sloan — COMMENTED** [review]
1 more tiny comment

**Review by srh-sloan — APPROVED** [review]
LGTM 👍

**Pull request merged**